src/solvers/smt2/smt2_conv.cpp

```diff
@@ -4869,8 +4869,10 @@ void smt2_convt::find_symbols_rec(
 
     if(recstack.find(id) == recstack.end())
     {
+      const auto &base_struct = ns.follow_tag(struct_tag);
       recstack.insert(id);
-      find_symbols_rec(ns.follow_tag(struct_tag), recstack);
+      find_symbols_rec(base_struct, recstack);
+      datatype_map[type] = datatype_map[base_struct];
```

Comment:
:pick: It would be nice to have some more doxygen at the declaration of `datatype_map` covering exactly what it maps from. It would also be nice to have better doxygen above `use_datatypes` documenting what changes should be expected in the output file when this is set to true. Obviously not a blocking comment, just wishful thinking.

Reply:
I'm adding some comments at `datatype_map` declaration, but I think I could make a separate PR for explaining `use_datatypes` better.

There's some documentation about `use_datatypes` sprinkled throughout `smt2_conv.cpp`, e.g. at https://github.com/diffblue/cbmc/blob/develop/src/solvers/smt2/smt2_conv.cpp#L4775-L4810. Probably we could add some high-level description above its declaration.